tls/extensions/s2n_client_key_share.c

```diff
@@ -156,6 +186,16 @@ static int s2n_client_key_share_recv(struct s2n_connection *conn, struct s2n_stu
 
         GUARD(s2n_ecc_evp_read_params_point(extension, share_size, &point_blob));
 
+        uint8_t *zeroes = NULL;
```

Comment:
How could the `malloc` and `free` be removed here? A variable length array could work, but should be memset on each iteration. Or maybe a single `s2n_alloc` using the largest supported curve before the while loop? We can `DEFER_CLEANUP` to free the blob.

Reply:
We are using a variable length array here as the length of the array changes with the share_size. We could refrain from being very precise and introduce only one allocation to the largest supported curve share_size.

Reply:
+1 on avoiding any heap allocations. If we need to do a zero check, a static global array sized to a reasonable max would work.

Regardless, I think we can remove this zero check altogether. See https://tools.ietf.org/html/rfc8446#section-4.2.8 

```
... However, the values MAY be a non-contiguous subset
   of the "supported_groups" extension and MAY omit the most preferred
   groups.  Such a situation could arise if the most preferred groups
   are new and unlikely to be supported in enough places to make
   pregenerating key shares for them efficient.
```
So there will be "holes" in the vector for the key_shares the client doesn't want to provide. The client does not express a missing key share as a zeroed out value, they simply don't write it in the vector.parsing/validation of course).

Reply:
Is this just to check if it's completely zeroed out? You could try something like https://godbolt.org/z/KcjyDV

Reply:
@raycoll  So the reason I went the route of checking for a vector of zeroes instead of NULL  is because for x25519, a NULL keyshare is considered a valid keyshare. Ref: https://moderncrypto.org/mail-archive/curves/2017/000896.html. 
And in this method, we want to ignore processing empty keyshares sent by the client.